Sources/MarathonCore/Export.swift

```diff
@@ -0,0 +1,123 @@
+
+public enum ExportError {
+    case missingPath
+    case noMarathonScriptFound
+    case failedToExportScript(String)
+}
+
+extension ExportError: PrintableError {
+    public var message: String {
+        switch self {
+        case .missingPath:
+            return "No script name/path given"
+        case .noMarathonScriptFound:
+            return "No marathon scripts found"
+        case .failedToExportScript(let name):
+            return "Failed to export script \(name)"
+        }
+    }
+    
+    public var hints: [String] {
+        switch self {
+        case .missingPath:
+            return ["Pass the name/path of a script file to export (for example 'marathon export myScript')"]
+        case .noMarathonScriptFound:
+            return ["Marathon can only export scripts managed by marathon"]
+        case .failedToExportScript:
+            return ["Remove directory or provide alternate export path"]
+        }
+    }
+}
+
+// MARK: - Task
+
+internal final class ExportTask: Task, Executable {
+    private typealias Error = ExportError
+    
+    func execute() throws {
+        
+        // Required argument given?
+        guard let scriptNameOrPath = arguments.first?.asScriptPath() else {
+            throw Error.missingPath
+        }
+        
+        // Script Name or Full Script Path given?
+        let isScriptName = scriptNameOrPath.components(separatedBy: "/").count == 1
+        
+        let file: File
+        if isScriptName {
+            // Does File exist in current directory?
+            file = try FileSystem().currentFolder.file(atPath: scriptNameOrPath)
+        } else {
+            // Does File exist at full path?
+            file = try FileSystem().createFile(at: scriptNameOrPath)
+        }
+        
+        // Is the File a script managed by marathon?
+        let isManagedScript = !(scriptManager.managedScriptPaths.filter { $0 == file.path }.isEmpty)
+        guard isManagedScript else {
+            throw Error.noMarathonScriptFound
+        }
+        
+        let exportFolder: Folder
+        if arguments.count > 1 {
+            // If more than one argument, try to create Folder at export-path
+            exportFolder = try FileSystem().createFolderIfNeeded(at: arguments[1])
+        } else {
+            // Otherwise use default export path
+            exportFolder = FileSystem().currentFolder
+        }
+        
+        let exportPath = exportFolder.path.appending(file.name.asScriptName())
+        if exportFolder.containsSubfolder(named: file.name.asScriptName()) {
+            throw Error.failedToExportScript("\(file.name). Directory already exists at path: \(exportPath)") // TODO: How to handle this?
+        }
+        
+        try export(file, to: exportFolder)
+    }
+    
+    private func export(_ file: File, to folder: Folder) throws {
+        guard let sourcesDir = try? folder.createSubfolder(named: "Sources") else {
+            throw Error.failedToExportScript(file.name)
```

Comment:
Instead of having to throw this error in a lot of places, how about just calling `try` directly in here, and let it throw any error - then catch those errors at the call site? For example, this expression here would simply be:

```swift
let sourcesFolder = try folder.createSubfolder(named: "Sources")
```

and then at the call site:

```swift
try perform(export(file, to: exportFolder), orThrow: Error.failedToExportScript(file.name))
```